app/services/process_payment_intent.rb

```diff
@@ -35,8 +35,7 @@ def call!
 
     OrderWorkflow.new(order).next
 
-    if last_payment.can_complete?
-      last_payment.complete!
```

Comment:
so the OrderWorkflow completes it already, right?

Reply:
Yes and no! If the order is in payment state, `OrderWorkflow.new(order).next` will move it to completion, which processes the payments, and sets the payment to complete (if capturing was successful).

In the case that the order is already completed, `OrderWorkflow.new(order).next` will not actually process the payments correctly!

Reply:
I see. That's clearly a leaky abstraction.